platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/linking/InOrderLinkerTests.java

```diff
@@ -63,52 +68,65 @@ class InOrderLinkerTests {
 
     private FakeTime time;
 
+    private NodeId self_id = new NodeId(0);
+    private NodeId other_id = new NodeId(1);
+
     /**
      * Generates a mock event with the given parameters
      *
+     * @param selfId                the id of the event creator
      * @param selfHash              the hash of the event
      * @param selfGeneration        the generation of the event
-     * @param selfParentHash        the hash of the self parent of the event
-     * @param selfParentGeneration  the generation of the self parent of the event
-     * @param otherParentHash       the hash of the other parent of the event
-     * @param otherParentGeneration the generation of the other parent of the event
+     * @param selfBirthRound        the birthRound of the event
+     * @param selfParent            the self parent event descriptor
+     * @param otherParent           the other parent event descriptor
      * @param selfTimeCreated       the time created of the event
      * @return the mock event
      */
     private static GossipEvent generateMockEvent(
+            @NonNull final NodeId selfId,
             @NonNull final Hash selfHash,
             final long selfGeneration,
-            @Nullable final Hash selfParentHash,
-            final long selfParentGeneration,
-            @Nullable final Hash otherParentHash,
-            final long otherParentGeneration,
+            final long selfBirthRound,
+            @NonNull final EventDescriptor selfParent,
+            @NonNull final EventDescriptor otherParent,
             @NonNull final Instant selfTimeCreated) {
 
-        final EventDescriptor descriptor = mock(EventDescriptor.class);
-        when(descriptor.getHash()).thenReturn(selfHash);
-        when(descriptor.getGeneration()).thenReturn(selfGeneration);
+        final EventDescriptor selfDescriptor = new EventDescriptor(selfHash, selfId, selfGeneration, selfBirthRound);
 
         final BaseEventHashedData hashedData = mock(BaseEventHashedData.class);
         when(hashedData.getHash()).thenReturn(selfHash);
         when(hashedData.getGeneration()).thenReturn(selfGeneration);
-        when(hashedData.getSelfParentHash()).thenReturn(selfParentHash);
-        when(hashedData.getSelfParentGen()).thenReturn(selfParentGeneration);
-        when(hashedData.getOtherParentHash()).thenReturn(otherParentHash);
-        when(hashedData.getOtherParentGen()).thenReturn(otherParentGeneration);
+        when(hashedData.getBirthRound()).thenReturn(selfBirthRound);
+        if (selfParent != null) {
```

Comment:
`selfParent` and `otherParent` are asserted `@NonNull`, but nullity is still being checked

Reply:
updated to Nullable, thank you.